junit/src/test/java/cucumber/runtime/junit/FeatureRunnerTest.java

```diff
@@ -89,17 +93,67 @@ public void should_call_formatter_for_scenario_outline_with_two_examples_table_a
     }
 
     private RunNotifier runFeatureWithNotifier(CucumberFeature cucumberFeature) throws InitializationError {
+        FeatureRunner runner = createFeatureRunner(cucumberFeature);
+        RunNotifier notifier = mock(RunNotifier.class);
+        runner.run(notifier);
+        return notifier;
+    }
+
+    private FeatureRunner createFeatureRunner(CucumberFeature cucumberFeature) throws InitializationError {
         final RuntimeOptions runtimeOptions = new RuntimeOptions("-p null");
         final ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
         final ClasspathResourceLoader resourceLoader = new ClasspathResourceLoader(classLoader);
         final RuntimeGlue glue = mock(RuntimeGlue.class);
         final Runtime runtime = new Runtime(resourceLoader, classLoader, asList(mock(Backend.class)), runtimeOptions, new TimeService.Stub(0l), glue);
-        FeatureRunner runner = new FeatureRunner(cucumberFeature, runtime, new JUnitReporter(runtime.getEventBus(), false, new JUnitOptions(Collections.<String>emptyList())));
-        RunNotifier notifier = mock(RunNotifier.class);
-        runner.run(notifier);
-        return notifier;
+        return new FeatureRunner(cucumberFeature, runtime, new JUnitReporter(runtime.getEventBus(), false, new JUnitOptions(Collections.<String>emptyList())));
+    }
+
+
+    @Test
+    public void shouldPopulateDescriptionsWithStableUniqueIds() throws Exception {
```

Comment:
Most test method uses the non-idomatic form of using underscore, since some/many of us contributors find them easier to read (and in tests readability is even more important than elsewhere), see discussion at https://github.com/cucumber/cucumber-jvm/pull/302#issuecomment-5267175